src/error.cpp

```diff
@@ -2,230 +2,207 @@
-        { Exiv2::kerUnhandledXmpdatum,
-          N_("Unhandled Xmpdatum %1 of type %2") }, // %1=key, %2=value type
-        { Exiv2::kerUnhandledXmpNode,
-          N_("Unhandled XMP node %1 with opt=%2") }, // %1=key, %2=XMP Toolkit option flags
-        { Exiv2::kerXMPToolkitError,
-          N_("XMP Toolkit error %1: %2") }, // %1=XMP_Error::GetID(), %2=XMP_Error::GetErrMsg()
-        { Exiv2::kerDecodeLangAltPropertyFailed,
-          N_("Failed to decode Lang Alt property %1 with opt=%2") }, // %1=property path, %3=XMP Toolkit option flags
-        { Exiv2::kerDecodeLangAltQualifierFailed,
-          N_("Failed to decode Lang Alt qualifier %1 with opt=%2") }, // %1=qualifier path, %3=XMP Toolkit option flags
-        { Exiv2::kerEncodeLangAltPropertyFailed,
-          N_("Failed to encode Lang Alt property %1") }, // %1=key
-        { Exiv2::kerPropertyNameIdentificationFailed,
-          N_("Failed to determine property name from path %1, namespace %2") }, // %1=property path, %2=namespace
-        { Exiv2::kerSchemaNamespaceNotRegistered,
-          N_("Schema namespace %1 is not registered with the XMP Toolkit") }, // %1=namespace
-        { Exiv2::kerNoNamespaceForPrefix,
-          N_("No namespace registered for prefix `%1'") }, // %1=prefix
-        { Exiv2::kerAliasesNotSupported,
-          N_("Aliases are not supported. Please send this XMP packet to ahuggel@gmx.net `%1', `%2', `%3'") }, // %1=namespace, %2=property path, %3=value
-        { Exiv2::kerInvalidXmpText,
-          N_("Invalid XmpText type `%1'") }, // %1=type
-        { Exiv2::kerTooManyTiffDirectoryEntries,
-          N_("TIFF directory %1 has too many entries") }, // %1=TIFF directory name
-        { Exiv2::kerMultipleTiffArrayElementTagsInDirectory,
-          N_("Multiple TIFF array element tags %1 in one directory") }, // %1=tag number
-        { Exiv2::kerWrongTiffArrayElementTagType,
-          N_("TIFF array element tag %1 has wrong type") }, // %1=tag number
-        { Exiv2::kerInvalidKeyXmpValue,
-          N_("%1 has invalid XMP value type `%2'") }, // %1=key, %2=value type
-        { Exiv2::kerInvalidIccProfile,
-          N_("Not a valid ICC Profile") },
-        { Exiv2::kerTiffDirectoryTooLarge,
-          N_("tiff directory length is too large") },
-        { Exiv2::kerInvalidTypeValue,
-          N_("invalid type in tiff structure") },
-        { Exiv2::kerInvalidLangAltValue,
-          N_("Invalid LangAlt value `%1'") }, // %1=value
-        { Exiv2::kerInvalidMalloc,
-          N_("invalid memory allocation request") },
-        { Exiv2::kerCorruptedMetadata,
-          N_("corrupted image metadata") },
-        { Exiv2::kerArithmeticOverflow,
-          N_("Arithmetic operation overflow") },
-        { Exiv2::kerMallocFailed,
-          N_("Memory allocation failed")}
+    constexpr std::array<const char*, static_cast<size_t>(Exiv2::ErrorCode::kerErrorCount)> errList{
```

Comment:
Since we are now on C++17 we can just let the compiler use a deduction guide:
```suggestion
    constexpr std::array errList{
```
And the biggest benefit is that we now don't force the length of the array and thus can add a `static_assert` below like
```cpp
static_assert(errList.size() == static_cast<size_t>(Exiv2::ErrorCode::kerErrorCount), "errList needs to contain a error msg for every ErrorCode defined in error.hpp");
```
to enforce that we have an error msg for every defined ErrorCode

Reply:
Done!